src/DistributedMetadata/CatalogService.cpp

```diff
@@ -576,13 +605,26 @@ CatalogService::TableContainerPerNode CatalogService::buildCatalog(const NodePtr
 /// Merge a snapshot of tables from one host
 void CatalogService::mergeCatalog(const NodePtr & node, TableContainerPerNode snapshot)
 {
+    std::unique_lock guard{catalog_rwlock};
+
     if (snapshot.empty())
-    {
+    {   
```

Comment:
Since this function is already long and complex. Can we refactor out this segment to another function  like `void deleteCatalogForNode(...)` ?

Reply:
fixed